### PR TITLE
Allow memory cards to upload configurations to any valid machine

### PR DESCRIPTION
**Pull request by nBeebz**
https://github.com/AppliedEnergistics/Applied-Energistics-2/issues/4727

Tested transferring configurations between a cell workbench, storage bus, and import bus with and without capacity cards.

(I tried to squash commit history but it only made them stronger, sorry)

### CODE REVIEW

**Comment by yueh** [comment]
This is way to error prone in terms of accidentally applying a few setting to a completely different machine. E.g. applying the priority of an export bus to a drive by accidentally clicking it without any feedback about what has changed. Some for applying a config which needs capacity cards in one machine, but not in other and so potentially assigning inaccessible config slots.

There needs at least some sort of marking compatible machines like storage buses, formation planes and maybe the cell workbench into a single "storage machine" category. (Assuming it is actually possible and not cause new issues).
Basically anything where the config can be fully applied and not just partly.
